libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -597,6 +597,89 @@ string YulUtilFunctions::overflowCheckedIntSubFunction(IntegerType const& _type)
 	});
 }
 
+string YulUtilFunctions::overflowCheckedIntExpFunction(
+	IntegerType const& _type,
+	IntegerType const& _exponentType
+)
+{
+	solUnimplementedAssert(!_type.isSigned(), "");
+	solAssert(!_exponentType.isSigned(), "");
+	string functionName = "checked_exp_" + _type.identifier() + "_" + _exponentType.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return
+			Whiskers(R"(
+			function <functionName>(base, exponent) -> power {
+				base := <cleanupFunction>(base)
+				exponent := <exponentCleanupFunction>(exponent)
+				// 0**0 == 1
+				if iszero(exponent) { power := 1 leave }
+				if or(
+					lt(base, 2),
+					eq(exponent, 1)
+				) { power := base leave }
+
+
+				let max := <maxValue>
+
+				// small base optimizations
+				if eq(base, 2) {
+					if gt(exponent, 255) { revert(0, 0) }
+					power := exp(2, exponent)
+					if gt(power, max) { revert(0, 0) }
+					leave
+				}
+				// b**e < 2**256 <=> e * log(b) < 256 * log(2) <=> e < 256 * log(2) / log(b)
+				if or(
+					and(lt(base, 11), lt(exponent, 78)),
+					and(lt(base, 257), lt(exponent, 33)),
```

Comment:
```suggestion
					and(lt(base, 257), lt(exponent, 32)),
```

Because `256**32 = 2**256 = UINT256_MAX + 1`